frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -316,6 +316,10 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 		}
 
 		if v := st.Platform; v != "" {
+			// Mark whether the specified platform is redudant for the linter
+			// to pick up later.
+			ds.platformRedundant = v == "$TARGETPLATFORM"
```

Comment:
This can also be "${}". Probably better to check the `platMatch`, so that the return value equals the targetplatform value and `platMatch.Matched` equals `TARGETPLATFORM`.

Reply:
I've modified this to use the platform match and integrated it more closely with the existing report unused args so I've removed this variable and this check.